sway/desktop/idle_inhibit_v1.c

```diff
@@ -104,10 +104,10 @@ void sway_idle_inhibit_v1_user_inhibitor_destroy(
 
 bool sway_idle_inhibit_v1_is_active(struct sway_idle_inhibitor_v1 *inhibitor) {
 	switch (inhibitor->mode) {
-	case INHIBIT_IDLE_APPLICATION:
+	case INHIBIT_IDLE_APPLICATION:;
 		// If there is no view associated with the inhibitor, assume visible
-		return !inhibitor->view || !inhibitor->view->container ||
-			view_is_visible(inhibitor->view);
+		struct sway_view *view = view_from_wlr_surface(inhibitor->wlr_inhibitor->surface);
```

Comment:
Can `inhibitor->wlr_inhibitor->surface` be NULL here?

Reply:
The idle inhibitor itself is destroyed on wlr_surface destroy, and the surface is not nulled anywhere, so unless inhibitors can be created with NULL surfaces somehow, the inhibitor surface pointer should be valid up until and including the inhibitor destroy signal.

Reply:
Makes sense!